powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/VaultUnlockServiceBehavior.java

```diff
@@ -162,16 +165,18 @@ public VaultUnlockResponse unlockVault(VaultUnlockRequest request) throws Generi
                 throw localizationProvider.buildExceptionForCode(ServiceError.INVALID_REQUEST);
             }
             // Lookup the activation
-            final ActivationRecordEntity activation = repositoryCatalogue.getActivationRepository().findActivationWithoutLock(activationId);
+            final Optional<ActivationRecordEntity> activationOptional = activationQueryService.findActivationWithoutLock(activationId);
 
             // Check if the activation is in correct state
-            if (activation == null || !ActivationStatus.ACTIVE.equals(activation.getActivationStatus())) {
+            if (activationOptional.isEmpty() || !ActivationStatus.ACTIVE.equals(activationOptional.get().getActivationStatus())) {
```

Comment:
If already touched, we may get rid of `equals` and use `!=` for enum.

Reply:
I will fix it in all occurrences.